Fix tests on MacOS (darwin)

ziogaschr opened the pull request:
Error is "The ipc endpoint is longer than %d characters."

ziogaschr committed:
cmd/geth: fix on darwin, where long test names fail

Error is "The ipc endpoint is longer than %d characters."

ziogaschr commented on cmd/geth/consolecmd_cg_test.go:
```diff
@@ -165,7 +167,10 @@ func TestGethStartupLogs(t *testing.T) {
 	}
 	for i, c := range cases {
-		t.Run(fmt.Sprintf("TestGethStartupLogs/%d: %v", i, c.flags), func(t *testing.T) {
+		// fix for darwin, where long test names fail with "The ipc endpoint is longer than %d characters."
```
@meowsbits another way is to change the test name only `if rpc.max_path_size < len([]byte(strings.Join(c.flags, " "))))`, for this we have to export the `max_path_size` in `rpc`.

meowsbits replied:
https://unix.stackexchange.com/questions/367008/why-is-socket-path-length-limited-to-a-hundred-chars

If the error you're seeing is because of the IPC path length limit, then we can try to shorten that up by shortening the `datadir` path, since geth should use the default `<datadir>/<alt chain?>/geth/geth.ipc`.

But if the error is just because of the test names, I recommend reducing them to a simple `Sprintf("%d", i)` -- which seems like a reasonable thing to do anyway (shoving the flags in there was a bad idea). 

I'm going to push a commit, see whatcha think.

ziogaschr replied:
The issue was because of the IPC path, which was long as part of it was based on test name.
I am going to test your commits

ziogaschr committed:
go.mod,go.sum: update dependency for etclabscore/go-openrpc-reflect@fix-m1-nil-pointer for fixing test issue

ziogaschr committed:
cmd/geth: fix lint issue (G401: Use of weak cryptographic primitive (gosec))

meowsbits committed:
cmd/geth: fixup subtest names, logging, and datadir paths

If IPC paths are too long, Unix systems will complain
because there are limits, albeit inconsistent ones.
So we try to keep IPC path names short (ie <100 chars).

I removed all instances where flags were used
in test names because
- they cause issues with apple silicon M1,
- and they are pointless

I also removed the redundant subtest naming scheme
which included the test name. Go includes the
test name by default anyway.

Date: 2022-08-29 09:21:05-05:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
go.mod,go.sum: bump etclabscore/go-openrpc-reflect to v0.0.37

Bumps dep to a tag version equivalent to the
previously referenced, untagged version.

Date: 2022-08-29 09:22:26-05:00
Signed-off-by: meows <b5c6@protonmail.com>
